contentcuration/contentcuration/frontend/channelEdit/components/QuickEditModal/EditAudienceModal.vue

```diff
@@ -0,0 +1,171 @@
+<template>
+
+  <KModal
+    :title="$tr('editAudienceTitle')"
+    :submitText="$tr('saveAction')"
+    :cancelText="$tr('cancelAction')"
+    data-test="edit-audience-modal"
+    @submit="handleSave"
+    @cancel="close"
+  >
+    <p data-test="resources-selected-message">
+      {{ $tr('resourcesSelected', { count: nodeIds.length }) }}
+    </p>
+    <template v-if="isMultipleAudience">
+      <p data-test="multiple-audience-message">
+        {{ $tr('multipleAudience') }}
+      </p>
+      <hr
+        :style="dividerStyle"
+      >
+    </template>
+    <h4 class="modal-subheading">
+      {{ $tr('visibleTo') }}
+    </h4>
+    <div
+      data-test="roles-options-list"
+    >
+      <KRadioButton
+        v-for="rol in rolesOptions"
+        :key="rol.value"
+        v-model="selectedRol"
+        data-test="rol-radio-button"
+        :label="rol.label"
+        :value="rol.value"
+        :description="rol.description"
+      />
+    </div>
+    <hr
+      :style="dividerStyle"
+    >
+    <KCheckbox
+      :checked="forBeginners"
+      data-test="for-beginners-checkbox"
+      :label="$tr('forBeginnersCheckbox')"
+      @change="value => forBeginners = value"
+    />
+  </KModal>
+
+</template>
+
+
+<script>
+
+  import { mapGetters, mapActions } from 'vuex';
+  import { RolesList, RolesNames } from 'shared/leUtils/Roles';
+  import { ResourcesNeededTypes } from 'shared/constants';
+  import { constantsTranslationMixin } from 'shared/mixins';
+
+  export default {
+    name: 'EditAudienceModal',
+    mixins: [constantsTranslationMixin],
+    props: {
+      nodeIds: {
+        type: Array,
+        required: true,
+      },
+    },
+    data() {
+      return {
+        selectedRol: '',
+        forBeginners: false,
+      };
+    },
+    computed: {
+      ...mapGetters('contentNode', ['getContentNodes']),
+      nodes() {
+        return this.getContentNodes(this.nodeIds);
+      },
+      rolesOptions() {
+        return RolesList.map(role => ({
+          value: role,
+          label: this.translateConstant(role),
+          description: this.rolesDescription[role],
+        }));
+      },
+      dividerStyle() {
+        return {
+          border: 0,
+          borderBottom: `1px solid ${this.$themeTokens.fineLine}`,
+          margin: '24px 0',
+        };
+      },
+      rolesDescription() {
+        return {
+          [RolesNames.COACH]: this.$tr('visibleToCoaches'),
+          [RolesNames.LEARNER]: this.$tr('visibleToAnyone'),
+        };
+      },
+      isMultipleAudience() {
+        const languages = new Set(this.nodes.map(node => node.role_visibility).filter(Boolean));
+        if (languages.size > 1) {
+          return true;
+        }
+        const forBeginners = new Set(
+          this.nodes.map(node =>
+            Boolean(node.learner_needs && node.learner_needs[ResourcesNeededTypes.FOR_BEGINNERS])
```

Comment:
I think we could just loop through the nodes and perform an early return whenever the condition `(node.learner_needs && node.learner_needs[ResourcesNeededTypes.FOR_BEGINNERS]` is met. This could potentially save on unnecessary loop cycles for larger nodes?

Reply:
Yes, I think so. Done :tada: